src/platforms/esp32/components/avm_builtins/uart_driver.c

```diff
@@ -186,18 +214,23 @@ Context *uart_driver_create_port(GlobalContext *global, term opts)
     term rx_pin = get_uart_pin_opt(opts, RX_ATOM);
     term rts_pin = get_uart_pin_opt(opts, RTS_ATOM);
     term cts_pin = get_uart_pin_opt(opts, CTS_ATOM);
+    if (UNLIKELY(tx_pin < -1 || rx_pin < -1 || rts_pin < -1 || cts_pin < -1)) {
```

Comment:
`get_uart_pin_opt` returns an int but here it's cast to a term and compared with an int.
This is highly suspicious and dirty. We should instead distinguish `UART_PIN_NO_CHANGE` which is defined from esp-idf's uart.h from an error code, for example by using an int *ok or by returning a boolean.

Reply:
Indeed! I overlooked that bug, these are ultimately used as int, they never should have been cast to a term.